lib/money_column/active_record_hooks.rb

```diff
@@ -48,7 +48,12 @@ def write_money_attribute(column, money)
         return self[column] = nil
       end
 
-      currency_raw_source = options[:currency] || (send(options[:currency_column]) rescue nil)
+      currency_raw_source = begin
+        options[:currency] || send(options[:currency_column]) || money.currency
```

Comment:
Unrelated to the order of determining the currency source, it would be nice to clean up this very broad `rescue`. From what I can tell only `NoMethodError` could raise here, and we can avoid leaning on this crutch for controlling flow without too much hassle.

Something like:
```ruby
currency_raw_source = options[:currency] || 
  (send(options[:currency_column]) if respond_to(options[:currency_column])) || 
  money.currency
```
WDYT?

Reply:
That's an assignment with an `OR` followed by a trailing `if` which itself contains an `OR` not to mention using send and respond to just feel funny to me. 
That's some real gymnastic code ha